Rework CLI installation docs (#638)
dbt-labs/docs.getdbt.com: website/docs/dbt-cli/install/from-source.md

---
title: "Install from source"
id: "from-source"
---

dbt Core is open source software, as are almost all adapter plugins. As such, the codebases are freely available to download and build from source. To do this, you can clone the repositories from GitHub and install the local version using `pip`.

> This is also an important part of contributing. If you'd like to help fix a bug or implement a sought-after feature. For more details, read the [core contributing guide](https://github.com/dbt-labs/dbt/blob/HEAD/CONTRIBUTING.md).

Reviewer: Those first two sentences feel a bit disjointed

```suggestion
You will also need to follow this process if you'd like to contribute to the project by fixing a bug or implementing a sought-after feature. For more details, read the [core contributing guide](https://github.com/dbt-labs/dbt/blob/HEAD/CONTRIBUTING.md).
```

Author: love it!

### Core

To install from source:

```shell

> git clone https://github.com/dbt-labs/dbt.git

Reviewer: ```suggestion
git clone https://github.com/dbt-labs/dbt-core.git
```

cd dbt
pip install -r requirements.txt
```

This will install `dbt-core` and `dbt-postgres`. To install in editable mode (includes your local changes as you make them), use `pip install -e editable-requirements.txt` instead.

### Plugin

> 
> To install a plugin from source:

Reviewer: Do you need to install core _and_ the adapter, or do adapters bring core in with them? We should clarify either way. The suggestion below assumes you need both.

Since it doesn’t _strictly_ need to be before, I'd be OK with "in addition to installing your plugin", but in practice everyone will do core before the adapter I suspect.

```suggestion

:::info

You need to install dbt-core as described above before installing your plugin. 

:::

To install a plugin from source:
```

```shell
git clone https://github.com/dbt-labs/dbt-redshift.git

> cd dbt-redshift

Reviewer: Everywhere else we've been using the generic case of `dbt-<adapter>` - any reason for the change?

Author: just because adapters live in many different repos. I'll say more here

---
title: "Install from source"
id: "from-source"
---

> dbt Core is open source software, as are almost all adapter plugins. As such, the codebases are freely available to download and build from source. To do this, you can clone the repositories from GitHub and install the local version using `pip`.

Reviewer: >as are almost all adapter plugins

Which ones aren't OSS? How would someone even install those?

Author: yeah, to be clear - if they're not OSS, we (probably) don't know about them! and the only people who can install them are the people who built them, or their colleagues. but, in theory, it's possible that there exist closed-source adapters (and we know of a handful).

You will also need to follow this process if you'd like to contribute to the project by fixing a bug or implementing a sought-after feature. For more details, read the [core contributing guide](https://github.com/dbt-labs/dbt/blob/HEAD/CONTRIBUTING.md).

### Core

To install from source:

```shell
git clone https://github.com/dbt-labs/dbt-core.git

> cd dbt

Reviewer: ```suggestion
cd dbt-core
```
